crates/bevy_ecs/src/query/iter.rs

```diff
@@ -1475,11 +1474,8 @@ impl<'w, 's, D: QueryData, F: QueryFilter, const K: usize> QueryCombinationIter<
     /// references to the same component, leading to unique reference aliasing.
     ///.
     /// It is always safe for shared access.
+    #[inline]
```

Comment:
I don't think this will have an effect since it's on a private item? AFAIUI the `#[inline]` hint matters mainly across compilation units, if we aren't inlined here it's probably because we're spilling over some other heuristic.

Reply:
Yeah, this seems to validate that this won't do much: https://matklad.github.io/2021/07/09/inline-in-rust.html

Reply:
For private functions, if you don't explicitly use the `inline ` keyword, the decision is left to LLVM, which handles most cases effectively(Whether to inline is depend on function complexity and size after inlined). However, in this case, it appears that LLVM does not inline the function on my machine. 

consider following code:
![image](https://github.com/user-attachments/assets/23dfaf11-99a2-4b2a-a460-51a0077f357b)

### without inline
![image](https://github.com/user-attachments/assets/963aa4e8-a037-413a-8c71-a77fb10175f9)



### with inline
![image](https://github.com/user-attachments/assets/3da28b52-5881-4f3e-a788-65539d308412)

Reply:
Okay, interesting, thanks for providing the proof, that's good enough for me!